runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +232,39 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageInfo(PackageInfo packageInfo,
+            String algorithm, String fingerprint) {
+        if (packageInfo.signatures == null) {
+            Log.d(TAG, "No signature in package info");
```

Comment:
My bad for saying "fatal error", I think we're saying the same thing differently. My point here is that failures in this method should be reported with `Log.e` instead of `Log.d` since they are more serious than regular informative debugging messages.

The only thing I didn't get is why a verification failure would just mean the runtime hasn't been released; the code is not likely to change once it is, and both now and then if the verification fails it should mean something very fishy is going on and the runtime should not be trusted, no?